src/dftbp/dftb/mdftb.F90

```diff
@@ -0,0 +1,1575 @@
+
+  !> Internal status of TMultiExpan.
+  type TMdftb
+    integer :: nAtoms, nSpecies, mOrb, nOrb, nSpin
+
+    !> Hubbard U values for atoms
+    real(dp), allocatable :: hubbu(:)
+
+    !> Species of atoms
+    integer, allocatable :: species(:), nOrbSpecies(:)
+    integer, allocatable :: onDQOCharges(:,:)
+
+    !> coordinates of the atom
+    real(dp), allocatable :: coords(:,:)
+
+    real(dp), allocatable :: atomicDIntgrl(:,:,:,:)
+    real(dp), allocatable :: atomicQIntgrl(:,:,:,:,:)
+
+    !> Mulliken charge per atom
+    real(dp), allocatable :: deltaMAtom(:)
+    !> Dipole charge per atom
+    real(dp), allocatable :: deltaDAtom(:,:)
+    !> Quadrupole charge per atom
+    real(dp), allocatable :: deltaQAtom(:,:,:)
+
+    !> evaluated for the E, Atom1, Atom2 at each geometry step
+    real(dp), allocatable :: f10AB(:,:,:)
+    real(dp), allocatable :: f20AB(:,:,:,:)
+    real(dp), allocatable :: f30AB(:,:,:,:,:)
+    real(dp), allocatable :: f40AB(:,:,:,:,:,:)
+    !> evaluated for the gradient, Atom1, Atom2 at each geometry step
+    real(dp), allocatable :: f50AB(:,:,:,:,:,:,:)
+
+    !> add for the H and the E
+    real(dp), allocatable :: pot10x1Atom(:)
+    real(dp), allocatable :: pot20x2Atom(:)
+    real(dp), allocatable :: pot10x0Atom(:,:)
+    real(dp), allocatable :: pot11x1Atom(:,:)
+    real(dp), allocatable :: pot21x2Atom(:,:)
+    real(dp), allocatable :: pot20x0Atom(:,:,:)
+    real(dp), allocatable :: pot21x1Atom(:,:,:)
+    real(dp), allocatable :: pot22x2Atom(:,:,:)
+    !> add for the gradient
+    real(dp), allocatable :: pot30x0Atom(:,:,:,:)
+    real(dp), allocatable :: pot31x1Atom(:,:,:,:)
+    real(dp), allocatable :: pot32x2Atom(:,:,:,:)
+
+    !> total energy
+    real(dp) :: energyTT, energyMD, energyDD, energyMQ, energyDQ, energyQQ
+
+  contains
+    procedure :: updateCoords
+    procedure :: pullDeltaDQAtom
+    procedure :: pushDeltaDQAtom
+    procedure :: updateDeltaDQAtom
+    procedure :: updateDQPotentials
+    procedure :: addMultiExpanHamiltonian
+    procedure :: addMultiExpanEnergy
+    procedure :: addMultiExpanGradients
+    procedure :: addAtomicDipoleMoment
+    procedure :: addAtomicQuadrupoleMoment
+    procedure :: getMultiExpanInfo
+    procedure :: getOrbitalEquiv
+  end type TMdftb
+
+  !> Number of dipole components used in tblite library (x, y, z)
+  integer, parameter :: dimDipole = 3
+
+  !> Number of quadrupole components used in tblite library (xx, xy, yy, xz, yz, zz)
+  integer, parameter :: dimQuadrupole = 6
+
+contains
+
+  !> Get information on required multipolar contributions
+  subroutine getMultiExpanInfo(this, nDipole, nQuadrupole)
+
+    !> Data structure
+    class(TMdftb), intent(in) :: this
+
+    !> Number of dipole moment components
+    integer, intent(out) :: nDipole
+
+    !> Number of quadrupole moment components
+    integer, intent(out) :: nQuadrupole
+
+    integer, parameter :: dimDipole = 3
+    integer, parameter :: dimQuadrupole = 6
+
```

Comment:
Please remove, they are only shadowing the corresponding module variables with the same content.